src/components/Export.tsx

```diff
@@ -100,157 +107,202 @@ const ExportDialog: ExportDialogComponent = ({
-                          results:
-                            typeof results === "function" ? "dynamic" : results,
-                        },
-                        message: error.message,
-                        stack: error.stack,
-                        version: process.env.VERSION,
+                        activeExportType,
+                        filename,
+                        results:
+                          typeof results === "function" ? "dynamic" : results,
                       },
-                    }).catch(() => {});
-                    setDialogOpen(true);
-                    setError((e as Error).message);
-                  } finally {
-                    updateExportProgress({ progress: 0, id: exportId });
-                  }
-                }, 1);
-              }}
-              style={{ minWidth: 64 }}
-              disabled={loading}
-            />
-          </div>
+                      message: error.message,
+                      stack: error.stack,
+                      version: process.env.VERSION,
+                    },
+                  }).catch(() => {});
+                  setDialogOpen(true);
+                  setError((e as Error).message);
+                } finally {
+                  updateExportProgress({ progress: 0, id: exportId });
+                }
+              }, 1);
+            }}
+            style={{ minWidth: 64 }}
+            disabled={loading}
+          />
         </div>
+      </div>
+    </>
+  );
+
+  const SendToPanel = (
+    <div className={Classes.DIALOG_FOOTER}>
+      <div className={Classes.DIALOG_FOOTER_ACTIONS}>
+        <Button text={"Cancel"} intent={Intent.NONE} onClick={onClose} />
+        <Button
+          text={"Send To Current Canvas"}
+          intent={Intent.PRIMARY}
+          onClick={() => {
+            typeof results === "object"
+              ? tempAddToCurrentCanvas(results)
+              : null;
+            setDialogOpen(false);
+          }}
+          style={{ minWidth: 64 }}
+          disabled={loading}
+        />
+      </div>
+    </div>
+  );
+
+  const tempAddToCurrentCanvas = (r: Result[]) => {
+    r.map((r) => {
+      document.dispatchEvent(
+        new CustomEvent("roamjs:query-builder:action", {
+          detail: {
+            action: "canvas",
+            uid: r.uid,
+            val: r.text,
+            queryUid: parentUid,
+          },
+        })
+      );
+    });
+  };
+
+  return (
+    <>
+      <Dialog
+        isOpen={dialogOpen}
+        canEscapeKeyClose={false}
+        canOutsideClickClose={false}
+        isCloseButtonShown={false}
+        title={`Export Query Results`}
+        autoFocus={false}
+        enforceFocus={false}
+        portalClassName={"roamjs-export-dialog-body"}
+      >
+        <Tabs id="export-tabs" large={true}>
+          <Tab id="sendto" title="Send To" panel={SendToPanel} />
+          <Tab id="export" title="Export" panel={ExportPanel} />
+        </Tabs>
```

Comment:
This is the most relevant change in `Export.tsx`, along with passing `parentUid` to export

Most of the add/delete is from moving ExportPanel into its own component.